src/api/Feed.js

```diff
@@ -0,0 +1,1022 @@
+            if (!this.isDestroyed()) {
+                if (this.hasError) {
+                    errorCallback(sortedFeedItems);
+                } else {
+                    successCallback(sortedFeedItems);
+                }
+            }
+        });
+    }
+
+    /**
+     * Sort valid feed items, descending by created_at time.
+     *
+     * @param {Array<?Comments | ?Tasks | ?FileVersions>} args - Arguments list of each item container
+     * type that is allowed in the feed.
+     * @return {Array<?Comments | ?Tasks | ?FileVersions>} the sorted feed items
+     */
+    sortFeedItems(...args: Array<?Comments | ?Tasks | ?FileVersions>): FeedItems {
+        const feedItems = [];
+        args.forEach((itemContainer) => {
+            if (itemContainer) {
+                feedItems.push(...itemContainer.entries);
+            }
+        });
+        return feedItems.sort((a, b) => Date.parse(b.created_at) - Date.parse(a.created_at));
+    }
+
+    /**
+     * Fetches the comments for a file
+     *
+     * @return {Promise} - the file comments
+     */
+    fetchComments(): Promise<?Comments> {
+        this.commentsAPI = new CommentsAPI(this.options);
+        return new Promise((resolve) => {
+            this.commentsAPI.offsetGet(
+                this.id,
+                resolve,
+                (err) => {
+                    this.errorCallback(err);
+                    resolve();
+                },
+                DEFAULT_START,
+                DEFAULT_END,
+                COMMENTS_FIELDS_TO_FETCH,
+                true
+            );
+        });
+    }
+
+    /**
+     * Fetches the versions for a file
+     *
+     * @return {Promise} - the file versions
+     */
+    fetchVersions(): Promise<?FileVersions> {
+        this.versionsAPI = new VersionsAPI(this.options);
+        return new Promise((resolve) => {
+            this.versionsAPI.offsetGet(
+                this.id,
+                resolve,
+                (err) => {
+                    this.errorCallback(err);
+                    resolve();
+                },
+                DEFAULT_START,
+                DEFAULT_END,
+                VERSIONS_FIELDS_TO_FETCH,
+                true
+            );
+        });
+    }
+
+    /**
+     * Fetches the tasks for a file
+     *
+     * @return {Promise} - the feed items
+     */
+    fetchTasks(): Promise<?Tasks> {
+        this.tasksAPI = new TasksAPI(this.options);
+        const requestData = {
+            params: {
+                fields: TASKS_FIELDS_TO_FETCH.toString()
+            }
+        };
+
+        return new Promise((resolve) => {
+            this.tasksAPI.get({
+                id: this.id,
+                successCallback: (tasks) => {
+                    this.fetchTaskAssignments(tasks).then(resolve);
+                },
+                errorCallback: (err) => {
+                    this.errorCallback(err);
+                    resolve();
```

Comment:
Should these instead call `reject`?

Reply:
It was intentional to call resolve. Since I am using Promise.all above, a reject would immediately resolve the promise and not let the partial state be there